### ISSUE TITLE
`reportUnusedFunction` should not consider tests

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#540]
```py
def _f(): ...

def test_f():
    _f()
```

i would expect this to report that `f()` is unused, maybe this should be configurable? would need a way to specify what a test is

**Comment by DetachHead** [comment]
it should be configurable, and when running in "application" mode (ie. not a library), it should also report unused public functions as well as not count usages in tests. this functionality would also need to rely on #311 i guess
